crates/rspack_core/src/build_chunk_graph/code_splitter.rs

```diff
@@ -798,7 +799,14 @@ Or do you want to use the entrypoints '{name}' and '{runtime}' independently on
           }
         })
         .filter_map(|dep_id| self.compilation.module_graph.dependency_by_id(&dep_id))
-        .collect()
+        .collect();
+      // keep the dependency original order if it does not have span, or sort the dependency by
+      // the error span
+      filtered_dep.sort_by(|a, b| match (a.span(), b.span()) {
```

Comment:
How webpack keep the order? Is webpack keep the order by `getOutgoingConnections`, can we move this into `get_outgoing_connections`

Reply:
Because the webpack module creation is single-threaded, the outer connection is ordered by default.